src/modules/10-common/hooks/useGetEntityMetadata.tsx

```diff
@@ -124,9 +129,27 @@ export const getPipelineUrl = async (scope: EntityScope, identifier: string, isN
     ? routesV2.toPipelines({ orgIdentifier, projectIdentifier, accountId: accountIdentifier, mode: NAV_MODE.ALL })
     : routesV1.toPipelines({ orgIdentifier, projectIdentifier, accountId: accountIdentifier })
 
+  const pipelineUrl = isNewNav
+    ? `${routesV2.toPipelineStudio({
+        accountId: accountIdentifier,
+        orgIdentifier,
+        projectIdentifier,
+        pipelineIdentifier: identifier,
+        mode: module ? undefined : NAV_MODE.ALL,
+        module
+      })}`
+    : `${routesV1.toPipelineStudio({
+        accountId: accountIdentifier,
+        orgIdentifier,
+        projectIdentifier,
+        pipelineIdentifier: identifier,
+        module
+      })}`
+
   if (pipelineMetadataResponse?.status !== 'SUCCESS') {
+    setPipelineSummaryError?.(pipelineMetadataResponse)
     // Without metadata can not open pipelineStudio so redirecting to list
-    return Promise.resolve(pipelineListUrl)
+    return Promise.resolve(pipelineUrl)
```

Comment:
```suggestion
    return Promise.reject(pipelineMetadataResponse)
```

Can we do Promise.reject instead and then directly use `setPipelineSummaryError` in the parent .catch block instead of passing the setter to this common hook